### PR TITLE
fix(bins): cut a divider's wall cutout from the divider's own top

### LINKED ISSUE #4108: Bin Dividers Don't Have Correct Wall Cutout Shape (Rounded Corners)
**Describe the bug**
When I add drawer dividers to a bin, the dividers have cutouts of the same overall size as set at the bin level, but they don't have the correct shape. 

**To Reproduce**
Steps to reproduce:
1. Create a bin
2. Add Wall Cutout (I tested with U-Shape and Scoop) with corner rounding
3. Add interior divider
4. Observe inconsistent corner radius and cutout shape in the dividers

**Expected behavior**
I expect the shape of the wall cutout to be consistent across the internal dividers

**Screenshots**
Image

Note rounded corners on first divider. The behaviour is inconsistent sometimes there is no corner rounding on any dividers, sometimes just the first one is incorrect.

**Environment:**

- Browser: Chrome 151
- Device: Windows PC

### PR DESCRIPTION
**Pull request by andymai**
Fixes #4108.

## What was wrong

An interior-divider cutout was built in the outer wall's vertical frame. Two things came from the wrong datum:

- **The rim.** The shoulder round-over is drawn tangent to the material's top face, and the cut was told that face was the top of the stacking lip — 4.4mm above where a divider ends on the multi-cavity path, 5.1mm on the divider-wall path. A 5mm blend therefore spends its whole radius in the air above the divider and reaches its top face 0.036mm wide. On the same bin the same setting rounds an outer wall visibly, which is exactly the report: the dividers have the right size and the wrong shape, inconsistently so, because whether anything lands at all depends on the radius clearing the lip.
- **The depth.** A percentage of the wall, applied from the wall's top. A divider shortened with `compartments.dividerHeight` took 13% of its height where 60% was asked for.

## The fix

The window is built in the divider's own frame: its top face is the rim, and depth is a percentage of what stands above the cavity floor. The blend then lands tangent to the surface being looked at.

Depth also comes off `binFloorMm` rather than `wallThickness` now. They differ on a spec base (2mm floor under a 1.2mm wall), and a 100% cut used to carry on past the divider and slot the floor.

The two divider paths do not end at the same height — the multi-cavity cut leaves pockets that reach the rim, while divider-wall boxes stop at the interior height or wherever a shortened divider stops — so `interiorDividerTopZ` resolves which one built them, and the feature's cache key carries the answer (`v1` → `v2`).

## Verification

Measured on generated bins, in `binGenerator.scenario.wallCutoutCorners.test.ts`. The divider's top is read off the divider and the floor off a compartment's own floor, so the assertions do not restate the constant chain under test.

At 0.3 / 1 / 3mm outboard of the cut edge, the divider's top face drops by the quarter arc's own value (3.29 / 2.00 / 0.42mm) and is untouched past the radius. Before the fix all three are 0.00. The 3mm probe is what separates an arc from a chamfer, which would take 2mm there.

The shortened-divider case asserts the cut takes the depth percentage of what stands above the floor: 3.60mm of a 6.00mm divider. Before the fix it took 0.88mm.

Also run: `wallCutouts` scenario + export, `compartments`, `mergedCompartments`, `tilted-dividers`, `overhangDividers`, `dividerSeat`, `split-wall-cutout-lip`, `split-cutout-shoulder`, `featureBuilder`, and the two heavy `dividerPatterns` files (the only catalog scenario with an interior cutout enabled) — 103 tests, no snapshot movement.

@ewing-digital thanks for the report, and for noting that it was inconsistent rather than simply absent — that inconsistency is what pointed at the lip.

https://claude.ai/code/session_0129Vusm2xfPbqEgLrFskyWi

<a href="https://cubic.dev/pr/andymai/gridfinity-layout-tool/pull/4111?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: src/features/generation/worker/generators/binGenerator.scenario.wallCutoutCorners.test.ts [hunk 1]
```diff
@@ -160,3 +167,145 @@ describe('wall cutout rounded shoulders', () => {
...
+
+/**
+ * The same round-over, on an interior divider.
+ *
+ * A divider has no stacking lip over it, so its own top face is the rim its
+ * shoulder blend is tangent to. Cut against the wall's rim instead — 4.4mm
+ * higher on a bin with a lip — a 5mm blend reaches the divider 0.036mm wide,
+ * which is what "the dividers don't have the correct shape" looked like.
+ *
+ * Every number here is read off the mesh rather than restated from the
```
[the remark this answers is not in the document]
Comment (andymai) [reply src/features/generation/worker/generators/binGenerator.scenario.wallCutoutCorners.test.ts:180]: Trimmed in 2a07844d13 — the report quote is gone and the blocks state the mechanism rather than the symptom. Keeping the two figures that are load-bearing: 0.036mm is why these bins carry a lip where the ones above deliberately do not, and the arc-vs-chamfer numbers are what the probe depths were chosen to separate.

File: src/features/generation/worker/generators/binGenerator.scenario.wallCutoutCorners.test.ts [hunk 2]
```diff
@@ -160,3 +167,145 @@ describe('wall cutout rounded shoulders', () => {
...
+  it('leaves the two bins the same size', () => {
+    const a = boundingBox(square.vertices);
+    const b = boundingBox(rounded.vertices);
+    for (const k of ['minX', 'maxX', 'minY', 'maxY', 'minZ', 'maxZ'] as const) {
+      expect(b[k], k).toBeCloseTo(a[k], 3);
+    }
+  });
+
+  it('rounds the divider shoulder tangent to the divider top', () => {
+    // Quarter arc centred a radius inboard and a radius down from the corner:
```
[the remark this answers is not in the document]
Comment (andymai) [reply src/features/generation/worker/generators/binGenerator.scenario.wallCutoutCorners.test.ts]: Right, the word was backwards — the centre sits a radius INTO the standing material, not inboard of it. Reworded in 2a07844d13.

File: src/features/generation/worker/generators/wallCutoutBuilder.ts [hunk 3]
```diff
@@ -314,7 +327,7 @@ export function computeInteriorDividerCutouts(
     // otherwise percentage of it.
     const cutW =
       cfg.widthMm !== null ? Math.min(cfg.widthMm, seg.wallLen) : seg.wallLen * (cfg.width / 100);
-    const cutH = interiorH * (cfg.depth / 100);
+    const cutH = dividerH * (cfg.depth / 100);
```
[the remark this answers is not in the document]
Comment (andymai) [reply src/features/generation/worker/generators/wallCutoutBuilder.ts:330]: Not taking this one. `pnpm run check:doc-drift` puts every doc on a size ratchet and CLAUDE.md asks for new detail to go into a matching `.claude/skills/*/SKILL.md` rather than a README, since READMEs are read far more often than they are needed. The divider-specific datum and the baked-vs-additive split are stated where they are enforced: `interiorDividerTopZ` names both paths, and the scenario test measures the divider's top on the divider rather than restating the chain. Happy to add it to the geometry-generation skill if it turns out to cost a debugging round.
